### DISCUSSION TITLE
Reset/Reload form on submit?

### DISCUSSION THREAD

**Discussion by dwgreen1** [formio/formio.js#5014] [Q&A] [answered]
I have a situation where a user is doing data entry and needs to submit several entries in a row. It would help if, after submit, the form would reset so that it's ready for the next form entry. Out of the box it doesn't do this. Any advice on how to accomplish this?

Up until now I've been using the simple form embed, but I'm happy to switch that up if it's necessary.

**Comment by brendanbond** [comment, marked as answer]
Hey @dwgreen1, thanks for the question. What I do in this case is to hook into the submit event, clear the submission, and refresh the form, something like:
```
Formio.createForm(formioElement, urlOrFormJson).then((form) => {
  form.on("submit", () => {
    form.submission = {};
    form.resetValue();
    form.refresh();
  });
});
```

**Reply by dwgreen1** [reply]
@brendanbond that ```form.submission = {};``` was one of the missing pieces for me. Thank you!

Our of curiosity, if I wanted to continue using the quick inline embed, is there an easy way to add an ```addEventListener``` with those three functions?

**Reply by brendanbond** [reply]
@dwgreen1 glad to hear it!

> Out of curiosity, if I wanted to continue using the quick inline embed, is there an easy way to add an addEventListener with those three functions?

Not really, because quick line embed is designed to be a one-stop shop for very very quickly and easily embedding a form without any framework or additional javascript. It's really just a script tag with a query string that loads your form and some additional libraries. You could potentially access your form via the global Formio object, but if you're doing that you might as well embed it using normal JS.

**Reply by dwgreen1** [reply]
Makes sense. Thank you!
